cloudflare/src/endpoints/dns.rs

```diff
@@ -167,6 +167,21 @@ pub enum DnsContent {
     SRV { content: String },
 }
 
+/// Type of the DNS record, along with a optional value
+/// Similar to [`DnsContent`], but is only used in [`ListDnsRecordsParams`]
+#[serde_with::skip_serializing_none]
+#[derive(Serialize, Clone, Debug)]
+#[serde(tag = "type")]
+pub enum DnsRecordTypeWithOptionalContent {
```

Comment:
Because it (`type`) is an optional parameter as shown in the documentation?

Reply:
> Because it (`type`) is an optional parameter as shown in the documentation?

I believe you mean the `content` parameter instead of `type`. If a `Some` value is used in the `content` field then it is added in addition to the `type` parameter when serializing the `record_type` field of `ListDnsRecordsParams` due to `#[serde(flatten)]`. It may be cleaner to instead add a separate `content` field to `ListDnsRecordsParams` so it can be specified without also specifying `type`.

Reply:
Then we won't be able to ensure the type of `content` matches with the type of `type`.